metrics/src/main/java/org/aerogear/mobile/metrics/MetricsRegistry.java

```diff
@@ -0,0 +1,35 @@
+package org.aerogear.mobile.metrics;
+
+import org.aerogear.mobile.metrics.interfaces.MetricsProvider;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * The metrics registry is the central storage for all metrics providers
+ */
+public final class MetricsRegistry {
+    private static MetricsRegistry INSTANCE;
+
+    private final Set<MetricsProvider> providers = new HashSet<>();
```

Comment:
[Optional] Set does not guarantee order. Not sure if order here will matter.

Reply:
don't think the order matters here, but i'm open for other opinions